gutenberg.php

```diff
@@ -16,20 +16,144 @@
+	define( 'GUTENBERG_NEXT_UNSUPPORTED_WP_VERSION', '6.5' );
+}
+
+if ( defined( 'ABSPATH' ) ) {
+	gutenberg_pre_init();
+} else {
+	die( 'Invalid request.' );
+}
 
 /**
- * Display a version notice and deactivate the Gutenberg plugin.
+ * Display a "WordPress version is too old" notice.
  *
  * @since 0.1.0
  */
-function gutenberg_wordpress_version_notice() {
+function gutenberg_wordpress_version_too_old_notice() {
 	echo '<div class="error"><p>';
-	/* translators: %s: Minimum required version */
-	printf( __( 'Gutenberg requires WordPress %s or later to function properly. Please upgrade WordPress before activating Gutenberg.', 'gutenberg' ), '5.9' );
+
+	if ( current_user_can( 'update_plugins' ) ) {
+		printf(
+			/* translators: %s: Minimum required version */
+			__( 'Gutenberg requires WordPress %s or later to function properly. Please upgrade WordPress.', 'gutenberg' ),
+			esc_html( GUTENBERG_MIN_WP_VERSION )
+		);
+	} else {
+		printf(
+			/* translators: %s: Minimum required version */
+			__( 'Gutenberg requires WordPress %s or later to function properly. Please ask an administrator to upgrade WordPress.', 'gutenberg' ),
+			esc_html( GUTENBERG_MIN_WP_VERSION )
+		);
+	}
+
 	echo '</p></div>';
+}
+
+/**
+ * Display a "Gutenberg version is too old" notice.
+ *
+ * @since 11.7.0
+ */
+function gutenberg_version_too_old_notice() {
+	echo '<div class="error"><p>';
+
+	if ( current_user_can( 'update_plugins' ) ) {
+		printf(
+			/* translators: %s: URL to the Plugins screen */
+			__( 'The Gutenberg plugin cannot be used. It is too old for your version of WordPress. Please <a href="%s">turn auto-updates on</a> or update it manually.', 'gutenberg' ),
+			admin_url( 'plugins.php' )
+		);
+	} else {
+		_e( 'The Gutenberg plugin cannot be used. It is too old for your version of WordPress. Please ask an administrator to update it.', 'gutenberg' );
+	}
+
+	echo '</p></div>';
+}
+
+/**
+ * Add a "WordPress version is too old" plugins list table notice.
+ *
+ * @since 11.7.0
+ *
+ * @param string[] $links Array of plugin action links.
+ * @param string   $file  Path to the plugin file relative to the plugins directory.
+ * @return string[] Updated array of plugin action links.
+ */
+function gutenberg_wordpress_version_too_old_action_links_notice( $links, $file ) {
+	$plugin_basename = basename( __DIR__ ) . '/gutenberg.php';
+
+	if ( $file === $plugin_basename && current_user_can( 'manage_options' ) ) {
+		// Prevent PHP warnings when a plugin uses this filter incorrectly.
+		$links = (array) $links;
+
+		// Can only be a very short notice.
+		$text = __( 'WordPress is too old. Gutenberg cannot be used.', 'gutenberg' );
+
+		// The 'button-link-delete' class name makes the text red.
+		$links['wp-gutenberg-plugin-notice button-link-delete'] = esc_html( $text );
+	}
+
+	return $links;
+}
+
+/**
+ * Add a "Gutenberg version is too old" plugins list table notice.
+ *
+ * @since 11.7.0
```

Comment:
This PR has been open for quite some time. These number will need to be updated before landing 😄

Reply:
Yep, still going through it at the moment. Should have a better/cleaned up version by tonight.